### ISSUE TITLE
Option completion doesn't work for commands

### ISSUE THREAD

**Issue by tschaub** [bcoe/yargs#180]
Given this `test.js` script:
```js
#!/usr/bin/env node
var argv = require('./index')
  .completion('completion')
  .command('bar', 'bar command', function(yargs) {
    var subArgv = yargs
      .option('foo', {
        describe: 'foo option'
      })
      .argv;

    console.log('subArgv: %j', subArgv);
  })
  .argv;

console.log('argv: %j', argv);
```

Enabling completion:
```bash
$ eval "$( ./test.js completion )"
```

And typing `./test.js bar --f` doesn't result in the `foo` option being completed.

**Comment by bcoe** [comment]
The parsing inside the command's handler is facilitated by a fresh instance of yargs. It would take a bit of work for to make completion work contextually, based on the command entered prior to the tab completion being hit. 

One could make the argument that this might be specific enough that we should just endorse that you implement it via a custom completion handler (the optional closure you can provide to completion).

To argue the other side though, it would be a slick feature to have command-specific completion -- is this something you'd be interested in taking a stab at?

**Comment by tschaub** [comment]
@bcoe yeah, I'd be interested in tackling it.  I may not have time for it until early next week.  So if anybody else is game, please have at it.  I'll leave word here when I start on it in earnest.

**Comment by tschaub** [comment]
Had a bit of time during a layover to look into this.  It may end up being pretty straightforward: https://github.com/bcoe/yargs/compare/master...tschaub:command-completion

This only works for commands with a handler that calls `completion` on the `yargs` object that it is passed.  For example:
```js
#!/usr/bin/env node
var argv = require('./index')
  .help('help')
  .completion()
  .command('foo', 'foo command', function(yargs) {
    var sub = yargs
      .options({
        bar: {
          describe: 'bar option'
        }
      })
      .completion()
      .help('help')
      .argv;

    process.exit(0);
  })
  .command('bam', 'bam command', function(yargs) {
    var sub = yargs
      .options({
        baz: {
          describe: 'baz option'
        }
      })
      .completion()
      .help('help')
      .argv;

    process.exit(0);
  })
  .argv
```

Needs testing and potentially more work.  I'll try to pick it up later on.

**Comment by bcoe** [comment]
cool \o/ I was worried that would be much more of a pain.

I'm also flying today, and am going to see about squashing a few more bugs on yargs. 

Question, how do you feel about the default completion handler also completing file names in the current working directory? I feel like this is something you usually want, but lose when you enable yargs' completion. Should we default to supporting this, or should the onus be on the developer to add this if they need it?

**Comment by tschaub** [comment]
@bcoe - I've opened #185 for command option completion (as noted there, this only works for commands one level deep).  This is working well for the use cases I've tried.  I'm sure it could be enhanced if there are more complex use cases.

> how do you feel about the default completion handler also completing file names in the current working directory

That could be nice.  One possible implementation in #186.  Could also do `fs.readdir(path.dirname(current), callback)` and spit out joined paths, but I think it might be more straightforward to delegate to `compgen`.
